src/headlessMoneyHash.ts

```diff
@@ -699,17 +704,24 @@ export default class MoneyHashHeadless<TType extends IntentType> {
    * ```
    *
    */
+  renderUrl(url: string, renderStrategy: UrlRenderStrategy): Promise<void>;
+  renderUrl(
+    url: string,
+    renderStrategy: UrlRenderStrategy,
+    options?: RenderOptions,
+  ): Promise<void>;
```

Comment:
for better overloading we only want options for REDIRECT

![Screenshot 2024-08-19 at 11 18 29 AM](https://github.com/user-attachments/assets/b3589373-0aaf-4bcb-92ac-2f64a5cce954)


```ts
  renderUrl(
    url: string,
    renderStrategy: Exclude<UrlRenderStrategy, "REDIRECT">,
  ): Promise<void>;
  renderUrl(
    url: string,
    renderStrategy: "REDIRECT",
    options?: RenderOptions,
  ): Promise<void>;
  async renderUrl(
    url: string,
    renderStrategy: unknown,
    options?: RenderOptions,
  ) {
    switch (renderStrategy) {
      case "IFRAME":
        return this.#renderUrlInIframe(url);
      case "POPUP_IFRAME":
        return this.#renderUrlInPopUpIframe(url);
      case "REDIRECT":
        return this.#renderUrlInRedirect(url, options);
      default:
        return null;
    }
  }
```